plugins/janus_nosip.c

```diff
@@ -1762,41 +1761,21 @@ char *janus_nosip_sdp_manipulate(janus_nosip_session *session, janus_sdp *sdp, g
 }
 
 /* Bind local RTP/RTCP sockets */
-static int janus_nosip_allocate_local_ports(janus_nosip_session *session) {
+static int janus_nosip_allocate_local_ports(janus_nosip_session *session, gboolean update) {
 	if(session == NULL) {
 		JANUS_LOG(LOG_ERR, "Invalid session\n");
 		return -1;
 	}
 	/* Reset status */
-	if(session->media.audio_rtp_fd != -1) {
-		close(session->media.audio_rtp_fd);
-		session->media.audio_rtp_fd = -1;
-	}
-	if(session->media.audio_rtcp_fd != -1) {
-		close(session->media.audio_rtcp_fd);
-		session->media.audio_rtcp_fd = -1;
-	}
-	session->media.local_audio_rtp_port = 0;
-	session->media.local_audio_rtcp_port = 0;
-	session->media.audio_ssrc = 0;
-	if(session->media.video_rtp_fd != -1) {
-		close(session->media.video_rtp_fd);
-		session->media.video_rtp_fd = -1;
-	}
-	if(session->media.video_rtcp_fd != -1) {
-		close(session->media.video_rtcp_fd);
-		session->media.video_rtcp_fd = -1;
-	}
-	session->media.local_video_rtp_port = 0;
-	session->media.local_video_rtcp_port = 0;
-	session->media.video_ssrc = 0;
-	if(session->media.pipefd[0] > 0) {
-		close(session->media.pipefd[0]);
-		session->media.pipefd[0] = -1;
-	}
-	if(session->media.pipefd[1] > 0) {
-		close(session->media.pipefd[1]);
-		session->media.pipefd[1] = -1;
+	if(!update) {
```

Comment:
I understand the usage of the new property, but why is the `!update` case only resetting `pipefd`, and not the audio and video RTP/RTCP stuff as well as we did before? Since this is the case of the call just starting, we need to be sure we're starting fresh. While we close those sockets and reset them when the relay thread closes, it's safer to do it here as well.

Reply:
As far as I understand current design the `!update` case is granted to be the first call to `_allocate` routine(). Please, correct me if I wrong. The idea was only to create `pipefd` once.
Even in your case, sockets and pipe remain opened if `relay_thread` has not been created for some reasons. Since you manage session life-time with reference counter, it is also might be safer to release these resources on session destroy.

Reply:
You can have multiple calls on the same handle in sequence, though. When you close a call, and then start a new one on the same handle (so not a renegotiation), you'll again hit the allocate method with a `!update`. Since you may have used audio and video in the previous call, we need to reset those sockets as well, as I don't want them to be reused if it's a different call.